notes.bash_completion

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+_notes_complete_notes() {
+    local configured_dir=${NOTES_DIRECTORY%/} # Remove trailing slashes
+    local notes_dir="${configured_dir:-$HOME/notes}"
+    local items=($(compgen -f "$notes_dir/$1"))
+    for item in "${items[@]}"; do
+        [[ -d $item ]] && item="$item/"
+        local filename=${item#$notes_dir/}
+        COMPREPLY+=("${filename%.md}")
+    done
+}
+
+_notes()
+{
+    local cur
+    readonly commands="new find grep open"
```

Comment:
This is very curious.

It seems to only become a problem the second time you press tab. This almost feels like a bug in bash to me, the variables should be scoped to the function yet they seem to survive multiple invocations, very strange.

I solved it by simply doing `local` instead, I will poke around a bit more to see if I can figure anything out.